stdlib/string.ncl

```diff
@@ -294,6 +295,9 @@
     | doc m%"
       `is_match regex str` checks if `str` matches `regex`.
 
+      Note that this function may perform better by sharing its partial application between multiple calls,
+      because in this case the underlying regular expression will only be compiled once.
+
```

Comment:
I think it deserves a small example, say:

```
let is_integer = fun kind x =>
  if kind == `Integer then
    string.is_match "[0-9]+" x
  else
    string.is_match "[0-9]*\\.?[0-9]+" x in
is_number `Integer "1" && is_number `Float "0.55"
```
vs
```
let is_integer = fun kind =>
  if kind == `Integer then
    string.is_match "[0-9]+"
  else
    string.is_match "[0-9]*\\.?[0-9]+" in
...
```

Or something like this (the code may need to be fixed, I haven't tried it).

Reply:
Would it make sense to include two examples (in both `is_match` and `match`)?